Formula/gptfdisk.rb

```diff
@@ -16,35 +16,43 @@ class Gptfdisk < Formula
 
   uses_from_macos "ncurses"
 
-  # update linker path for libncurses
-  patch :DATA
+  on_linux do
+    depends_on "util-linux"
+  end
 
   def install
-    system "make", "-f", "Makefile.mac"
+    on_macos do
+      inreplace "Makefile.mac" do |s|
+        s.gsub! "/usr/local/Cellar/ncurses/6.2/lib/libncurses.dylib", "-L/usr/lib -lncurses"
+        s.gsub! "-L/usr/local/lib -lpopt", "-L#{Formula["popt"].opt_lib} -lpopt"
+      end
+
+      system "make", "-f", "Makefile.mac"
+    end
+
+    on_linux do
+      deps = %w[ncurses popt util-linux]
+      ENV["CFLAGS"] = deps.map { |dep| "-I#{Formula[dep].opt_include}" }.join(" ")
+      ENV["CXXFLAGS"] = deps.map { |dep| "-I#{Formula[dep].opt_include}" }.join(" ")
+      ENV["LDFLAGS"] = deps.map { |dep| "-L#{Formula[dep].opt_lib}" }.join(" ")
+
+      system "make", "-f", "Makefile"
+    end
+
     %w[cgdisk fixparts gdisk sgdisk].each do |program|
       bin.install program
       man8.install "#{program}.8"
     end
   end
 
   test do
-    system "dd", "if=/dev/zero", "of=test.dmg", "bs=1m", "count=1"
+    blocksize = "1"
+    on_macos { blocksize << "m" }
+    on_linux { blocksize << "M" }
+
+    system "dd", "if=/dev/zero", "of=test.dmg", "bs=#{blocksize}", "count=1"
```

Comment:
Maybe just use POSIX-compliant `k` (https://pubs.opengroup.org/onlinepubs/007904975/utilities/dd.html) rather than GNU/BSD specific `M`/`m`:
```suggestion
    system "dd", "if=/dev/zero", "of=test.dmg", "bs=1024k", "count=1"
```

GNU supports it https://www.gnu.org/software/coreutils/manual/html_node/Block-size.html#Block-size as does BSD https://www.freebsd.org/cgi/man.cgi?dd(1)